src/main/kotlin/no/nav/familie/ef/personhendelse/inntekt/InntektsendringerService.kt

```diff
@@ -13,35 +13,55 @@ class InntektsendringerService(
-        val nyesteRegistrerteInntekt = inntektshistorikkResponse.inntektForMåned(YearMonth.now().minusMonths(1).toString())
-        val nestNyesteRegistrerteInntekt = inntektshistorikkResponse.inntektForMåned(YearMonth.now().minusMonths(2).toString())
+        val nyesteRegistrerteInntekt =
+            inntektshistorikkResponse.inntektForMåned(YearMonth.now().minusMonths(1).toString())
+        val nestNyesteRegistrerteInntekt =
+            inntektshistorikkResponse.inntektForMåned(YearMonth.now().minusMonths(2).toString())
 
-        return har10ProsentHøyereInntektEnnForventet(nestNyesteRegistrerteInntekt, forventetInntektForPerson.personIdent, forventetInntektForPerson.forventetInntektToMånederTilbake) &&
-            har10ProsentHøyereInntektEnnForventet(nyesteRegistrerteInntekt, forventetInntektForPerson.personIdent, forventetInntektForPerson.forventetInntektForrigeMåned)
+        val antallProsentToMånederTilbake = antallProsentHøyereInntektEnnForventet(
+            nestNyesteRegistrerteInntekt,
+            forventetInntektForPerson.personIdent,
+            forventetInntektForPerson.forventetInntektToMånederTilbake,
+        )
+        val antallProsentForrigeMåned = antallProsentHøyereInntektEnnForventet(
+            nyesteRegistrerteInntekt,
+            forventetInntektForPerson.personIdent,
+            forventetInntektForPerson.forventetInntektForrigeMåned,
+        )
+
+        return EndretInntekt(
+            antallProsentForrigeMåned >= 10 && antallProsentToMånederTilbake >= 10,
+            (antallProsentForrigeMåned + antallProsentToMånederTilbake) / 2,
+        )
     }
 
-    private fun har10ProsentHøyereInntektEnnForventet(nyesteRegistrerteInntekt: List<InntektVersjon>?, ident: String, forventetInntekt: Int?): Boolean {
+    private fun antallProsentHøyereInntektEnnForventet(nyesteRegistrerteInntekt: List<InntektVersjon>?, ident: String, forventetInntekt: Int?): Int {
         if (forventetInntekt == null || nyesteRegistrerteInntekt?.maxOfOrNull { it.versjon } == null) {
             secureLogger.warn("Ingen gjeldende inntekt funnet på person $ident har personen løpende stønad?")
-            return false
+            return 0
         }
-        if (forventetInntekt > 585000) return false // Ignorer alle med over 585000 i årsinntekt, da de har 0 i utbetaling.
+        if (forventetInntekt > 585000) return 0 // Ignorer alle med over 585000 i årsinntekt, da de har 0 i utbetaling.
         val månedligForventetInntekt = (forventetInntekt / 12)
 
         val orgNrToNyesteVersjonMap = nyesteRegistrerteInntekt.associate { it.opplysningspliktig to it.versjon }
-        val inntektListe = nyesteRegistrerteInntekt.filter { it.versjon == orgNrToNyesteVersjonMap.get(it.opplysningspliktig) && it.arbeidsInntektInformasjon.inntektListe != null }.flatMap { it.arbeidsInntektInformasjon.inntektListe!! }
-
+        val inntektListe = nyesteRegistrerteInntekt.filter {
```

Comment:
Hvordan fungerer `nyesteRegistrerteInntekt`? Og då spesielt 
`val orgNrToNyesteVersjonMap = nyesteRegistrerteInntekt.associate { it.opplysningspliktig to it.versjon }`

Finner det flere like `opplysningspliktig` i den listen?